common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/HaleSchemaUtil.java

```diff
@@ -33,4 +41,40 @@ public static InputStream getHaleSchemaXSD() {
 		return HaleSchemaUtil.class.getResourceAsStream("hsd.xsd");
 	}
 
+	/**
+	 * Combine the given schemas to a single schema.
+	 * 
+	 * @param schemas the schemas to combine
+	 * @param reporter the reporter if available
+	 * @return the combined schema
+	 */
+	public static Schema combineSchema(List<Schema> schemas, IOReporter reporter) {
+		if (schemas == null || schemas.isEmpty()) {
+			// empty schema
+			return new DefaultSchema(null, null);
+		}
+		else if (schemas.size() == 1) {
+			return schemas.get(0);
+		}
+		else {
+			DefaultSchema result = new DefaultSchema(null, null);
+
+			schemas.forEach(schema -> {
+				schema.getTypes().forEach(type -> {
+					if (reporter != null) {
+						// check if type is already there
+						QName name = type.getName();
+						if (result.getType(name) != null) {
+							reporter.warn(
+									MessageFormat.format("Multiple definitions of type {0}", name));
+						}
+					}
+					result.addType(type);
```

Comment:
I think that's definitely fine for the use case at hand. 

A possible problem that might arise in the future: if `combineSchema` is used in another context where `reporter` is `null`, this will silently use the last definition in case of multiple definitions. 

I think we can simply document that behaviour for the time being. If we want to guard against that case, we could throw an exception in the `reporter == null` case instead.